openedx_blockstore_relay/transfer_data.py

```diff
@@ -0,0 +1,233 @@
+        """
+        Return the content_type to set for the given block_key.
+        """
+        return 'olx/{}'.format(block_type)
+
+    def transfer_block_to_bundle(self):
+        """
+        Upload block-related files to the Bundle.
+        """
+        LOG.debug('Transfer %s to Bundle <%s>', self.block_key, self.bundle_uuid)
+        block = compat.modulestore().get_item(self.block_key)
+        self.upload_olx(block)
+        self.upload_assets()
+        self.upload_manifest()
+
+    def upload_olx(self, block):
+        """
+        Upload the OLX for the given block definition, recursing through children.
+        """
+        block_key = block.location
+        if block_key in self._block_olx_ok:
+            # Prevent infinite recursion: don't re-create OLX for blocks that we've already done.
+            return None
+
+        block_type = block_key.block_type
+        if block_type in ('course', 'chapter', 'sequential', 'vertical'):
+            # These blocks' definition_to_xml methods require a runtime, so we avoid calling them.
+            olx_node = ElementTree.Element(block_type)
+        else:
+            try:
+                filesystem = WrapFS(MemoryFS())
+                olx_node = block.definition_to_xml(filesystem)
+                self._upload_related_files(filesystem)
+            except (NotImplementedError, AttributeError):
+                LOG.warning("Block type not supported, skipping (%s)", block_key)
+                self._block_olx_ok[block_key] = False
+                return None
+            except ResourceNotFound as exc:
+                LOG.error("Error fetching block OLX (%s): %s", block_key, exc)
+                self._block_olx_ok[block_key] = False
+                return None
+
+        for child in block.get_children():
+            LOG.debug('Add child %s', child.location)
+            self.upload_olx(child)
+            if self._block_olx_ok[child.location]:
+                # If we've successfully processed this child's OLX,
+                # include a reference to the child in the parent block.
+                child_ref = ElementTree.Element(child.location.block_type)
+                child_ref.set('url_name', child.url_name)
+                olx_node.append(child_ref)
+
+        self._upload_olx(olx_node, block_key)
+        self._block_olx_ok[block_key] = True
+        return olx_node
+
+    def upload_assets(self):
+        """
+        Upload the collected static asset files to the bundle.
+        """
+        for (asset_path, asset_uri) in self.assets:
+            content = compat.get_asset_content_from_path(self.block_key.course_key, asset_path)
+            if content is None:
+                LOG.error("Static asset not found: (%s, %s)", asset_path, asset_uri)
+            else:
+                self._post_bundle_file(
+                    path=asset_path,
+                    data=content.data,
+                    content_type=content.content_type,
+                )
```

Comment:
Note to self: Add the uploaded asset bundle file path in the `bundle.js` manifest `assets` list.

Reply:
Done, cf d5eb6e4 .
@symbolist Can you confirm that I understood this correctly?